src/zjs_callbacks.c

```diff
@@ -388,6 +388,53 @@ void print_callbacks(void)
 #define print_callbacks() do {} while (0)
 #endif
 
+#define MAX_ERROR_NAME_LENGTH       32
+#define MAX_ERROR_MESSAGE_LENGTH    128
+
+static void print_error_message(jerry_value_t error)
+{
+    uint32_t size;
+    char* message = NULL;
+    jerry_value_t name_val = zjs_get_property(error, "name");
+    if (!jerry_value_is_string(name_val)) {
+        ERR_PRINT("error did not have name\n");
+        // we should never get here.
+        return;
+    }
+    jerry_value_t msg_val = zjs_get_property(error, "message");
+    if (!jerry_value_is_string(msg_val)) {
+        ERR_PRINT("error did not have message\n");
+        // we should never get here.
+        return;
+    }
+
+    size = MAX_ERROR_NAME_LENGTH;
+    char name[size];
+
+    zjs_copy_jstring(name_val, name, &size);
```

Comment:
When you don't explicitly care about the string size, you're better off calling zjs_alloc_from_jstring and passing it a 0 maxlen. This will allocate from the heap as long as there's enough memory to fit it. I think that suffices in most cases and it makes the calling code simpler. Just be sure to free it when you're done.

I guess you still need to check for NULL, which I might not have done in a couple places. O_o! So tell me if I'm wrong and it doesn't end up being easier. :)